controllers/vpn/wireguardserver_controller.go

```diff
@@ -0,0 +1,645 @@
+Endpoint = %s:%s"`, serverPublicKey, serverAddress, wireguard.Status.Port)
+		if peer.Status.Config != newConfig || peer.Status.Status != vpnv1alpha1.Ready {
+			peer.Status.Config = newConfig
+			peer.Status.Status = vpnv1alpha1.Ready
+			peer.Status.Message = "Peer configured"
+			if err := r.Status().Update(ctx, &peer); err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+func (r *WireguardServerReconciler) getWireguardPeers(ctx context.Context, req ctrl.Request) (*vpnv1alpha1.WireguardPeerList, error) {
+	peers := &vpnv1alpha1.WireguardPeerList{}
+	if err := r.List(ctx, peers, client.InNamespace(req.Namespace)); err != nil {
+		return nil, err
+	}
+
+	relatedPeers := &vpnv1alpha1.WireguardPeerList{}
+
+	for _, peer := range peers.Items {
+		if peer.Spec.WireguardRef == req.Name {
+			relatedPeers.Items = append(relatedPeers.Items, peer)
+		}
+	}
+
+	return relatedPeers, nil
+}
+
+func (r *WireguardServerReconciler) configmapForWireguard(wireguardServer *vpnv1alpha1.WireguardServer) *corev1.ConfigMap {
+	return &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      wireguardServer.Name + "-config",
+			Namespace: wireguardServer.Namespace,
+			Labels:    labelsForWireguard(wireguardServer.Name),
+		},
+	}
+}
+
+func (r *WireguardServerReconciler) deploymentForWireguard(s *vpnv1alpha1.WireguardServer, image string) *appsv1.Deployment {
+	ls := labelsForWireguard(s.Name)
+	replicas := int32(1)
+
+	return &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      s.Name + "-dep",
+			Namespace: s.Namespace,
+			Labels:    ls,
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: &replicas,
+			Selector: &metav1.LabelSelector{
+				MatchLabels: ls,
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: ls,
+				},
+				Spec: corev1.PodSpec{
+					Volumes: []corev1.Volume{
+						{
+							Name: "socket",
+							VolumeSource: corev1.VolumeSource{
+
+								EmptyDir: &corev1.EmptyDirVolumeSource{},
+							},
+						},
+						{
+
+							Name: "config",
+							VolumeSource: corev1.VolumeSource{
+								Secret: &corev1.SecretVolumeSource{
+									SecretName: s.Name,
+								},
+							},
+						}},
+					Containers: []corev1.Container{
+						{
+							SecurityContext: &corev1.SecurityContext{
+								Capabilities: &corev1.Capabilities{Add: []corev1.Capability{"NET_ADMIN"}},
+							},
+							Image:           "mindflavor/prometheus-wireguard-exporter:3.5.1",
+							ImagePullPolicy: "Always",
+							Name:            "metrics",
+							Command:         []string{"/usr/local/bin/prometheus_wireguard_exporter"},
+							Ports: []corev1.ContainerPort{
+								{
+									ContainerPort: metricsPort,
+									Name:          "metrics",
+									Protocol:      corev1.ProtocolTCP,
+								}},
+							VolumeMounts: []corev1.VolumeMount{
+								{
+									Name:      "socket",
+									MountPath: "/var/run/wireguard/",
+								},
+							},
+						},
```

Comment:
In the wireguard server spec we should have a `sidecars` field with type `[]corev1.Container{)` so users can define their own sidecars for the exporter for example.

Reply:
done